tests/components/goalzero/test_config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Test Goal Zero Yeti config flow."""
+from goalzero import exceptions
+
+from homeassistant.components.goalzero.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from . import (
+    CONF_CONFIG_FLOW,
+    CONF_DATA,
+    CONF_HOST,
+    CONF_NAME,
+    NAME,
+    _create_mocked_yeti,
+    _patch_config_flow_yeti,
+)
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+def _flow_next(hass, flow_id):
+    return next(
+        flow
+        for flow in hass.config_entries.flow.async_progress()
+        if flow["flow_id"] == flow_id
+    )
+
+
+def _patch_setup():
+    return patch(
+        "homeassistant.components.goalzero.async_setup_entry",
+        return_value=True,
+    )
+
+
+async def test_flow_user(hass):
+    """Test user initialized flow."""
+    mocked_yeti = await _create_mocked_yeti()
+    with _patch_config_flow_yeti(mocked_yeti), _patch_setup():
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"] == {}
+        _flow_next(hass, result["flow_id"])
```

Comment:
Why do we call this?

Reply:
Turns out to be unnecessary. Will be removed.